File: leetcode/enumerated/128.py
# https://leetcode.com/problems/longest-consecutive-sequence/

class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums = set(nums)

        longest = 0
        for num in nums:
            if num - 1 not in nums:
                current = 1
                while num + 1 in nums:
                    num += 1
                    current += 1

                longest = max(longest, current)

        return longest

File: leetcode/enumerated/cpp/128.cpp
// http://leetcode.com/problems/longest-consecutive-sequence

class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        std::set<int> numset(nums.begin(), nums.end());
        int result = 0;

        for (int i : numset) {
            if (!numset.contains(i - 1)) {
                int current = 1;
                while (numset.contains(i + 1)) {
                    i++;
                    current++;
                }

                result = std::max(current, result);
            }
        }

        return result;
    }
};
